website/source/docs/agent/options.html.md

```diff
@@ -281,7 +281,7 @@ The options below are all specified on the command-line.
   port to listen on. Default -1 (https disabled). See [ports](#ports)
   documentation for more detail.
 
-* <a name="_log_file"></a><a href="#_log_file">`-log-file`</a> - to redirect all the Consul agent log messages to a file. This can be specified with the complete path along with the name of the log. In case the path doesn't have the filename, the filename defaults to `consul-{timestamp}.log`. Can be combined with <a href="#_log_rotate_bytes"> -log-rotate-bytes</a> and <a href="#_log_rotate_duration"> -log-rotate-duration </a> for a fine-grained log rotation experience.
+* <a name="_log_file"></a><a href="#_log_file">`-log-file`</a> - write all the Consul agent log messages to a file. The value is used as a prefix for the log file name. The current timestamp is appended to the file name. If the value ends in a path separator, `consul-` is used as the file name prefix. If the file name is missing an extension, `.log` is appended. For example, setting `log-file` to `/var/log` would result in a log file path of `/var/log/consul-{timestamp}.log`. `log-file` can be combined with <a href="#_log_rotate_bytes"> -log-rotate-bytes</a> and <a href="#_log_rotate_duration"> -log-rotate-duration </a> for a fine-grained log rotation experience.
```

Comment:
My only nitpick is that it isn't in the same "person"(?) as the flag above it
Ex: ` write all the Consul agent log messages to a file. The value is used as a prefix for the log file name.`
would be: ` writes all the Consul agent log messages to a file. This value is used as a prefix for the log file name.`

Reply:
Good call out! I'm used to writing reference docs in a different style, but I agree consistent is important here. Pushed a new commit with that change!

Reply:
I would reformat this slightly. For example;

> The value here is used to help assemble a complete log path. The value here can take a fully qualified file path to a directory, or a file specifically. When a value is provided here, a completely unique log file is assembled.
> When the value is a directory, a consul log file will be created, named `consul-{timestamp}.log`. 
> When the value is a file, the path expects the string `consul` as a prefix, will automatically assign a timestamp, and appends .log to the file. An example is `"log_file" : "/var/log/consul/output"` will return a log file named `/var/log/consul/consul-output-{timestamp}.log`

Does this make sense?

Reply:
@s-christoff , can you confirm that functionality?

Reply:
@dnephin , @s-christoff and I talked about this. Here's the functionality demonstrated. The string **after** "=" is the log file after the command is ran.

```
consul agent -dev --log-file=/var/log/ = /var/log/consul-$timestamp.log
consul agent -dev --log-file=/var/log/output = /var/log/ouput-$timestamp.log
consul agent -dev --log-file=/var/log/output.txt = /var/log/output-$timestamp.txt
consul agent -dev --log-file=/var/log/output.blah.dotfriend.ok.txt = /var/log/output.blah.dotfriend.ok-$timestamp.txt
``` 

Here's how I would write this:

> `log_file` can take either a directory, or a fully qualified file path as values. In the case of specifying a directory, e.g. `/var/log/consul/`, the Consul agent will create a file named `consul-${timestamp}.log`. In the case of specifying a file, e.g. `/var/log/consul/output.txt`, the agent will create a file named `output-${timestamp}.txt`. The file path notation uses the last period as a delimiter. Everything before the period deliminter is used as a prefix, and anything after becomes the file extension. All Consul log files should be generated in [prefix]-[timestamp].[ext] notation.

@hashicorp/consul-docs please weigh in!

Reply:
I think unfortunately some of that is not true. The code does not actually check for a directory, it looks for a trailing path separator. If someone were to specify a path to a directory without a trailing slash they would get files in the parent directory. I just noticed my example is wrong for exactly this reason.

I've attempted to describe the behavior as it is, which is unfortunately difficult!

Do you think it is necessary to describe file extensions here? I had assumed that was widely used terminology, but if it is not, I can go into more detail.

I made some minor changes to fix the example and try to clarify how `consul-` is added to the value.

Reply:
`Do you think it is necessary to describe file extensions here? I had assumed that was widely used terminology, but if it is not, I can go into more detail.`
I think it is and you're safe in that aspect. While trying to write docs it may be a good to have a discussion on making this flag a bit more user friendly - but we can do that internally. :)